### PR TITLE
Check connection to Vmanage Server

### PR DESCRIPTION
**Pull request by danielui**
Check connection to Vmanage Server by ping before start session.

### CODE REVIEW

File: vmngclient/session.py [hunk 1]
```diff
@@ -66,8 +67,13 @@ def create_vManageSession(
         Session object
 
     """
-    session = vManageSession(url=url, username=username, password=password, port=port, subdomain=subdomain)
-    session.auth = vManageAuth(session.base_url, username, password, verify=False)
+    connection = ping(url, verbose=True)
+    if connection.stats_packets_returned == 0:
+        raise ConnectionError("Server is not available")
```
Comment (tehAgitto) [inline vmngclient/session.py:72]: it should be separate method

**Review by tehAgitto — CHANGES_REQUESTED** [review]

File: vmngclient/session.py (hunk 1 above)
Comment (danielui) [reply vmngclient/session.py]: ok, fixed

File: vmngclient/session.py [hunk 2]
```diff
@@ -66,8 +66,11 @@ def create_vManageSession(
         Session object
 
     """
+
```
Comment (tehAgitto) [inline vmngclient/session.py:69]: Remove empty line.

File: vmngclient/session.py [hunk 3]
```diff
@@ -107,6 +110,14 @@ def create_vManageSession(
     return session
 
 
+def check_vmanage_server_connection(url):
+    url = f"http://{url}"
```
Comment (tehAgitto) [inline vmngclient/session.py:114]: Use `__create_base_url` as its used in every method.

File: vmngclient/session.py [hunk 4]
```diff
@@ -107,6 +110,14 @@ def create_vManageSession(
     return session
 
 
+def check_vmanage_server_connection(url):
```
Comment (tehAgitto) [inline vmngclient/session.py:113]: Add unittest.

File: vmngclient/session.py [hunk 5]
```diff
@@ -107,6 +110,14 @@ def create_vManageSession(
...
+    url = f"http://{url}"
+    try:
+        head(url, timeout=2)
+    except ConnectionError:
+        logger.error("vManage server is not available")
```
Comment (tehAgitto) [inline vmngclient/session.py:118]: Should we reraise an exception?

**Review by tehAgitto — CHANGES_REQUESTED** [review]
Why do `poetry.lock` file  is being modified even if we do not update nor add any package?

File: vmngclient/session.py (hunk 2 above)
Comment (danielui) [reply vmngclient/session.py:69]: Fixed

File: vmngclient/session.py [hunk 6]
```diff
@@ -284,3 +288,12 @@ def __eq__(self, other):
             ]
             return True if all(comparison_list) else False
         return False
+
+    def check_vmanage_server_connection(self) -> None:
```
Comment (tehAgitto) [inline vmngclient/session.py:292]:
```suggestion
    def check_vmanage_server_connection(self) -> bool:
```

File: vmngclient/session.py (hunk 3 above)
Comment (danielui) [reply vmngclient/session.py]: Fixed

File: vmngclient/session.py (hunk 6 above)
Comment (danielui) [reply vmngclient/session.py]: Done

File: vmngclient/session.py (hunk 5 above)
Comment (danielui) [reply vmngclient/session.py]: Exception is being reraised now

**Review by sbasan — APPROVED** [review]
lgtm

**Review by tehAgitto — APPROVED** [review]

**Pull request merged**